library/src/main/java/com/novoda/downloadmanager/AndroidLocalFilesDirectory.java

```diff
@@ -0,0 +1,25 @@
+package com.novoda.downloadmanager;
+
+import android.content.Context;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class AndroidLocalFilesDirectory implements LocalFilesDirectory {
```

Comment:
You can avoid making this public if this class is constructed by a factory, then only the factory and the interface is exposed and you keep the implementation detail private

Reply:
We use it in the activity so it has to be public.

Reply:
Only a factory and the interface needs to be public, the specific implementation can be private. If you go through a factory+interface then you can evolve and change the specific implementation and the public API keeps being unchanged along with all the public classes.

`LocalFilesDirectory  localFilesDirectory = LocalFilesDirectoryFactory.create(...)` should be enough

or maybe I'm missing something?

Reply:
As far as I am aware you are completely correct @zegnus. You can expose a factory in the download manager that will return the interface `LocalFilesDirectory` hiding the concrete implementation. 💯